src/backend/distributed/metadata/dependency.c

```diff
@@ -160,6 +161,54 @@ static ViewDependencyNode * BuildViewDependencyGraph(Oid relationId, HTAB *nodeM
 static Oid GetDependingView(Form_pg_depend pg_depend);
 
 
+PG_FUNCTION_INFO_V1(citus_get_dependencies_for_object);
+
+
+/*
+ * citus_unmark_object_distributed(classid oid, objid oid, objsubid int)
+ *
+ * removes the entry for an object address from pg_dist_object. Only removes the entry if
+ * the object does not exist anymore.
+ */
+Datum
+citus_get_dependencies_for_object(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+
+	Oid classid = PG_GETARG_OID(0);
+	Oid objid = PG_GETARG_OID(1);
+	int32 objsubid = PG_GETARG_INT32(2);
+
+	TupleDesc tupleDescriptor = NULL;
+	Tuplestorestate *tupleStore = SetupTuplestore(fcinfo, &tupleDescriptor);
+
+	ObjectAddress address = { 0 };
+	ObjectAddressSubSet(address, classid, objid, objsubid);
+
+	List *dependencies = GetDependenciesForObject(&address);
```

Comment:
this skips already distributed objects. So, Even if I merge this PR. I'd need to play with it on local objects to see how Citus would behave

Reply:
would it make sense to have this function return _all_ objects, with an optional filter function, which could skip anything already filtered?

The name of the function is a bit misleading for the user that it skips already distributed objects. I can probably help you here if it makes sense.